jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -298,6 +298,64 @@ public EndPoint getEndPoint()
         return _endPoint;
     }
 
+    /**
+     * <p>Return the Local Name of the connected channel.</p>
+     *
+     * <p>
+     * This is the name after the connector is bound and the connection is accepted.
+     * The value is typically the local address and optionally the local host name
+     * if the local address is unavailable.
+     * </p>
+     * <p>
+     * Value can be overridden by {@link Connector#setLocalAuthority(HostPort)} for
+     * scenarios where Jetty is being an intermediary and the local name
+     * needs to be changed to satisfy public (pre-intermediary) HTTP behaviors
+     * such as absolute-URI creation (eg: Location response header).
+     * </p>
+     *
+     * @return the local name if overridden, or the local address, or
+     * null in the case of no local address (usually seen in connectors not based on IP networking).
+
+     */
+    public String getLocalName()
```

Comment:
Why convert to separate name/port methods at the channel level?
I think it would be better if the channel provided `getServerAuthority()` and `getLocalAuthority()` (or `getLocalAddress()` if name change above is accepted).    We only need separate name/port methods in the request where they are defined by servlet API

Reply:
Note that the only real reason I see to resolve to name/port methods here is that the existing `getLocalAddress()` returns an `InetSocketAddress`, whilst the new methods return `HostPort`.      I'm thinking it would be simpler to just use `InetSocketAddress` everywhere.